### PR TITLE
Remove and replace some stale [ActiveIssue(...)]

### PR DESCRIPTION
**Pull request by stephentoub**
Remove some stale [ActiveIssues] where the tests now pass.

Replace some others with [PlatformSpecific], where the behavior is deemed to be "by design" / accepted.

And update a few issue numbers.

### CODE REVIEW

File: src/System.Globalization/tests/CompareInfo/CompareInfoTests.Compare.cs [hunk 1]
```diff
@@ -320,7 +320,7 @@ public void Compare(CompareInfo compareInfo, string string1, int offset1, int le
         [ActiveIssue(5436, PlatformID.AnyUnix)]
```
Comment (bartonjs) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.Compare.cs]: 5436 is closed... and about TcpClient...

I think the comment was right, and this is wrong, 5463 is "Investigate failing System.Globalization.Tests on Unix"

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 2]
```diff
@@ -807,7 +807,7 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
         }
 
         [Fact]
-        [ActiveIssue(3565, PlatformID.OSX)]
+        [PlatformSpecific(~PlatformID.OSX)] // Channel binding on Unix is only supported with OpenSSL backed to libcurl
```
Comment (bartonjs) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Shouldn't this fail on RHEL/CentOS, then?

File: src/System.Globalization/tests/CompareInfo/CompareInfoTests.Compare.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Globalization/tests/CompareInfo/CompareInfoTests.Compare.cs]: Yup.

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Yes, I'll make it conditional.

Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Though, this ran and passed on centos in ci. I need to investigate why.

**Comment by stephentoub** [comment]
Test Innerloop Windows_NT Release Build and Test please (https://github.com/dotnet/corefx/issues/6255)

**Pull request merged**